### PR TITLE
HRRR data queue

### PR DESCRIPTION
**Pull request by scotthavens**
- had to add a `data_variables` to the hrrr_grib to load the correct variables into the data queue
- Add new tests for the Lakes to test different methods of running HRRR with loading first or for each timestep
- tests using the mixin to check netcdf files

### CODE REVIEW

File: smrf/data/hrrr_grib.py [hunk 1]
```diff
@@ -111,7 +115,7 @@ def load_timestep_thread(self, date_times, data_queue):
         for date_time in date_times[1:]:
             self.load_timestep(date_time)
 
-            for variable in self.variables:
+            for variable in self.data_variables:
```
Comment (scotthavens) [inline smrf/data/hrrr_grib.py:118]: This was the major issue. The `variables` were filtered to non-wind variables where the `data_queue` was expecting there to be values to get (even if they aren't used).

**Comment by scotthavens** [comment]
Testing locally, this fixes the errors seen in AWSM PR [#89](https://github.com/USDA-ARS-NWRC/awsm/pull/89)

File: smrf/data/hrrr_grib.py (hunk 1 above)
Comment (jomey) [reply smrf/data/hrrr_grib.py:118]: Do I understand this correct that the threaded solution will revert back to loading the data even though not used?

Comment (scotthavens) [reply smrf/data/hrrr_grib.py:118]: It's not loading wind so it keeps your PR #209. On line 78 it still filters out the wind variables so they aren't loaded. This just ensures that the `data_queue` is populated using the empty dataframes set in `InputGribHRRR.calculate_wind`

Comment (jomey) [reply smrf/data/hrrr_grib.py:118]: Got it. Mainly asking to make sure there is consistent behavior independent of running threaded or not.

**Review by jomey — APPROVED** [review]

**Pull request merged**